lib/src/clojure_lsp/refactor/transform.clj

```diff
@@ -228,16 +241,40 @@
         (recur next-loc (or marked? exists?))
         (edit/back-to-mark-or-nil bind' :first-occurrence)))))
 
-(defn find-let-form [zloc]
-  (some-> zloc
-          (edit/find-ops-up "let")
-          z/up))
+(defn find-let-form
+  "Finds a let-form that would be valid to move zloc to"
+  [zloc uri db]
+  (let [let-loc (some-> zloc
+                        (edit/find-ops-up "let")
+                        z/up)]
+    (when let-loc
+      (let [{:keys [col row end-row end-col]} (meta (z/node zloc))
+            clj-analysis (unify-to-one-language (:analysis @db))
+            bindings-loc (z/right (z/down (zsub/subzip let-loc)))
+            local-defs (->> (q/find-local-usages-under-form
```

Comment:
Usually this is not a problem but this function that is called all the time on code actions has increased it's complexity and it's probably slower than before, with a chance to slow code actions process as a whole, right?
Should we check performance of this new function?
Usually we just manually test both old and new code on a code snippet that should be relevant, WDYT?

Reply:
Good catch, unify to one language is expensive, and upon reflection I think we don't want to call it either way. We would want to invalidate the move if the local escapes in any language.

The other calls are pretty quick it seems to pop up immediately in large functions and in large files. But my client doesn't keep the actions up like I think emacs does. So I dropped the unify but I think the rest is safe since it operates on local-analysis.

Reply:
looks good, I was afraid regarding `q/find-local-usages-under-form` and especially `q/find-definition` since it maps over all var-definitions.

BTW emacs calculate the code actions each time user change a line but with smart caches.

Reply:
yup, fyi since the only bucket going into `find-definition` will be locals, the multi-method will only look at the file's local analysis.